### PR TITLE
fix release compilation

### PR DESCRIPTION
**Pull request by byrcolin**

### CODE REVIEW

File: Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp [hunk 1]
```diff
@@ -170,7 +170,9 @@ namespace OpenXRVk
 
     bool Device::AcquireSwapChainImageInternal(AZ::u32 viewIndex, XR::SwapChain* baseSwapChain)
     {
+#if !defined(RELEASE)
         SwapChain* swapChain = static_cast<SwapChain*>(baseSwapChain);
```
Comment (moudgils) [inline Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp:174]: This is a little messy. Can we just move this cast into the assert itself. Then we wont need this ifdef

Comment (byrcolin) [reply Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp]: done

File: Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp [hunk 2]
```diff
@@ -198,7 +197,7 @@ namespace OpenXRVk
         }
 
         AZ_Assert(m_viewCountOutput == viewCapacityInput, "Size mismatch between xrLocateViews %i and xrEnumerateViewConfigurationViews %i", m_viewCountOutput, viewCapacityInput);
-        AZ_Assert(m_viewCountOutput == swapChain->GetViewConfigs().size(), "Size mismatch between xrLocateViews %i and xrEnumerateViewConfigurationViews %i", m_viewCountOutput, swapChain->GetViewConfigs().size());
+        AZ_Assert(m_viewCountOutput == static_cast<SwapChain*>(baseSwapChain)->GetViewConfigs().size(), "Size mismatch between xrLocateViews %i and xrEnumerateViewConfigurationViews %i", m_viewCountOutput, swapChain->GetViewConfigs().size());
```
Comment (moudgils) [inline Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp:200]: swapchain is used twice in this assert. The second one is swapChain->GetViewConfigs().size()

Comment (byrcolin) [reply Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp]: fixed

**Review by moudgils — APPROVED** [review]

**Review by moraaar — APPROVED** [review]

**Pull request merged**